contracts/core/modules/CTokenExchangeIssuanceModule.sol

```diff
@@ -0,0 +1,439 @@
+            address cTokenAddress = cTokenAddresses[i];
+            address underlyingAddress = cTokenWhiteList.getAddressValueByKey(cTokenAddress);
+
+            // Add approvals of the underlying token to the cToken contract
+            ERC20Wrapper.approve(
+                underlyingAddress,
+                cTokenAddress,
+                CommonMath.maxUInt256()
+            );
+
+            // Add approvals of the underlying token to the transferProxy contract
+            ERC20Wrapper.approve(
+                underlyingAddress,
+                _transferProxy,
+                CommonMath.maxUInt256()
+            );
+
+            // Add approvals of the cToken to the transferProxy contract
+            ERC20Wrapper.approve(
+                cTokenAddress,
+                _transferProxy,
+                CommonMath.maxUInt256()
+            );
+        }
+    }
+
+    /* ============ Public Functions ============ */
+
+    /**
+     * Performs trades via exchange wrappers to acquire components, mints cToken from underlying, and issues a Set to the caller
+     *
+     * @param _exchangeIssuanceParams              A Struct containing exchange issuance metadata
+     * @param _orderData                           Bytes array containing the exchange orders to execute
+     */
+    function exchangeIssue(
+        ExchangeIssuanceLibrary.ExchangeIssuanceParams memory _exchangeIssuanceParams,
+        bytes memory _orderData
+    )
+        public
+        nonReentrant
+    {
+        // Ensures validity of exchangeIssuanceParams
+        validateExchangeIssuanceParams(_exchangeIssuanceParams);
+
+        // Validate that all receiveTokens are components of the Set
+        SetTokenLibrary.validateTokensAreComponents(
+            _exchangeIssuanceParams.setAddress,
+            _exchangeIssuanceParams.receiveTokens
+        );
+
+        // Transfer the send tokens to the appropriate exchanges
+        transferSendTokensToExchangeWrappers(
+            _exchangeIssuanceParams.sendTokenExchangeIds,
+            _exchangeIssuanceParams.sendTokens,
+            _exchangeIssuanceParams.sendTokenAmounts
+        );
+
+        // Calculate expected receive token balances
+        uint256[] memory requiredBalances = calculateReceiveTokenBalances(
+            _exchangeIssuanceParams
+        );
+
+        // Execute the exchange orders using the encoded order data and deposits tokens in vault
+        executeExchangeOrders(_orderData);
+
+        // Withdraw underlying tokens required from vault, mint cTokens in the module, and return to vault under sender address
+        mintCTokens(
+            _exchangeIssuanceParams.receiveTokens,
+            _exchangeIssuanceParams.receiveTokenAmounts
+        );
+
+        // Check that sender's receive tokens in Vault have been incremented correctly
+        ExchangeIssuanceLibrary.validatePostExchangeReceiveTokenBalances(
+            vault,
+            _exchangeIssuanceParams.receiveTokens,
```

Comment:
In this case, the receiveTokens are the cTokens or the components again?

Reply:
These are cTokens. For redeeming, the sendTokens are the cTokens